src/introducer.py

```diff
@@ -38,14 +41,24 @@ async def _vetting_loop(self):
                 return
             try:
                 log.info("Vetting random peers.")
-                rawpeers = self.global_connections.peers.get_peers(
-                    100, True, self.recent_peer_threshold
+                if self.global_connections.introducer_peers is None:
+                    await asyncio.sleep(3)
+                    continue
+                rawpeers = self.global_connections.introducer_peers.get_peers(
+                    100, True, 3 * self.recent_peer_threshold
                 )
 
                 for peer in rawpeers:
                     if self._shut_down:
                         return
-                    if peer.get_hash() not in self.vetted:
+                    if peer.get_hash() in self.vetted_timestamps:
+                        if time.time() > self.vetted_timestamps[peer.get_hash()] + 3600:
+                            if peer.get_hash() in self.vetted:
+                                self.vetted[peer.get_hash()] = False
+                    if (
```

Comment:
this will still try to vet entries that were constantly marked as False (since they still won't be in the map, but get_peers will still retrieve them). only if we give only 1 vet per hour for each peer (so at most 24 vets per day) this could work. is this what you mean?